_rules/visible-label-in-accessible-name-2ee8b8.md

```diff
@@ -97,10 +97,10 @@ This button has [visible][] text that is contained within the [accessible name][
 
 #### Passed Example 5
 
-This button has [visible][] text that does not need to be contained within the [accessible name][], because the "x" text node is [non-text content][].
+This button has [visible][] text that does not need to be contained within the [accessible name][], because the "x" text node is [non-text content][]. Note: this would need to meet SC 1.1.1 Non text content.
 
 ```html
-<button aria-label="close">X</button>
+<button aria-label="anything">X</button>
```

Comment:
Shouldn't we keep the "Close" label, which makes this pass 1.1.1?
(assuming this is indeed a close button)

Reply:
Because the Expectation says

> except for characters in the [text nodes][] used to express [non-text content][]

there is no expectation, and this example passes. I used `aria-label="anything"` to make it clear that really any description would pass this rule. But, you are right that SC 1.1.1 would then be not satisfied. But I'd like to consider SC 1.1.1 being secondary in a separate PR.

Reply:
This comes up often around here: when passed examples fail something else.  

I like it best when there is a note on the passed example which documents that "something else".  In this case that note could be: "This example passes the rule and will fail SC 1.1.1".

We could call these "curve ball passed examples". 

To have examples like these, and to document them as such, meets two goals, both important:
* It emphasizes _why_ the example passes the rule.  (In this case: "to make it clear that really any description would pass this rule", as Kathy put it).  I think that this aids understanding, a lot.
* It tells everyone that even though this is a "passed example" _for this rule_, it's not a good example overall, so no one should copy and paste the code.

Jean-Yves did this a lot in the [examples](https://github.com/act-rules/act-rules.github.io/blob/c302b7504ccc66acc2715fa5884d7d2411dbe45e/_rules/aria-role-permitted-j7zzqr.md) for his recent [PR 2084](https://github.com/act-rules/act-rules.github.io/pull/2084): 
* "Using an explicit role of generic for any element is globally not recommended, but not an error by itself."
* "Using an explicit role which is the same as the [implicit role](https://github.com/act-rules/act-rules.github.io/blob/c302b7504ccc66acc2715fa5884d7d2411dbe45e/_rules/aria-role-permitted-j7zzqr.md#implicit-role) is not recommended, but not an error by itself."
* "The btn token is not a valid role and is therefore ignored."
* "Even though the alert role is not allowed, the first token is a valid role and is therefore used."

I think it's a good thing

Reply:
This works, but perhaps a cleaner way to handle this would be to modify the second condition of the applicability to filter out not just non-visible text content, but also non-text content. That would cause this example to move to the non-applicable section.

Reply:
Agree, Tom. 

But, Applicability has to be objective at this time, so the _except non-text content_ part is under the Expectation section for now.